utils/jwt.ts

```diff
@@ -9,19 +9,51 @@ export function genJwt(id: string): string {
   return jwt.sign({id}, process.env.TOKEN_SECRET, { expiresIn: '7d' });
 }
 
-export function validateJwt(req: Request, res: Response, next: NextFunction) {
-  const authHeader = req.headers.authorization;
-  const token = authHeader && authHeader.split(' ')[1];
+interface validateJwtOptions {
+  req: Request;
+  res: Response;
+  next: NextFunction;
+  token?: string;
+  key: string;
+  tokenName: string;
```

Comment:
this extra field doesn't really make sense..

i feel like a better design is just to make the `key` act as the flag..